enhancements/cluster-logging/performance-tuning.md

```diff
@@ -0,0 +1,260 @@
+---
+title: performance-tuning
+authors:
+  - "@alanconway"
+reviewers:
+  - "@jcantrill"
+approvers:
+  - "@jcantrill"
+api-approvers:
+  - "@jcantrill"
+creation-date: 2023-12-15
+last-updated: 2023-12-15
+tracking-link:
+  - https://issues.redhat.com/browse/OBSDA-549
+see-also:
+replaces:
+superseded-by:
+---
+
+# Performance tuning
+
+## Summary
+
+A _performance tuning_ API to control performance, reliability and special protocol features of an output,
+without exposing the complexity of the underlying collector configuration.
+
+**Note**
+- Only vector is be supported initial, there are no current plans to back-port to fluentd.
+- Existing `output[].limits` rate limiting feature is separate from this proposal. The implementations may interact.
+
+## Motivation
+
+Performance and reliability tuning in the underlying collector configuration is complex and error prone.
+We want to expose a sufficient set of controls for realistic use-cases,
+but we don't want to expose the full configuration surface of Vector.
+
+### User Stories
+
+#### As a cluster logging administrator I want to minimize log loss.
+
+``` yaml
+outputs:
+ new - name: minimize-log-loss
+   tuning:
+     delivery: AtLeastOnce
+```
+
+#### As a cluster logging administrator I want to maximize throughput.
+
+``` yaml
+outputs:
+  - name: maximize-throughput
+    tuning:
+      delivery: AtMostOnce
+```
+
+#### As a cluster logging administrator I want to tweak features of a specific output.
+
+``` yaml
+outputs:
+  - name: detailed-tweaks
+    tuning:
```

Comment:
Needs update to match https://github.com/openshift/cluster-logging-operator/blob/master/api/logging/v1/cluster_log_forwarder_types.go#L234-L269